### PR TITLE
/api/info performance improvements

### PR DESCRIPTION
**Pull request by martastain**
This pull request introduces several changes to the `api/info` endpoint  The key changes include the addition of Redis caching, refactoring of site information retrieval, and improvements to the onboarding status checks.

### CODE REVIEW

File: api/system/info.py [hunk 1]
```diff
@@ -114,47 +136,75 @@ async def get_sso_options(request: Request) -> list[SSOOption]:
...
+    """
 
-        if user.name not in site.users:
+    async for row in Postgres.iterate(query, query_id, user_name):
+        site = SiteInfo(id=row["id"], **row["data"])
+        if current_site and site.id == current_site.id:
+            # record matches the current site
+            current_site_exists = True
+            if user_name not in site.users:
+                current_site.users.extend(site.users)
```
[the remark this answers is not in the document]
Comment (martastain) [reply api/system/info.py:165]: current_site (from the request) contains only the username invoking the request, so this won't cause duplicates. But it's maybe counterintuitive.

File: api/system/info.py [hunk 2]
```diff
@@ -114,47 +136,75 @@ async def get_sso_options(request: Request) -> list[SSOOption]:
...
+        WHERE id = $1 OR data->'users' ? $2
+    """
 
-        if user.name not in site.users:
+    async for row in Postgres.iterate(query, query_id, user_name):
+        site = SiteInfo(id=row["id"], **row["data"])
+        if current_site and site.id == current_site.id:
+            # record matches the current site
+            current_site_exists = True
+            if user_name not in site.users:
```
[the remark this answers is not in the document]
Comment (BigRoy) [reply api/system/info.py:164]: But if it's at most a few users - disregard this of course.

[the remark this answers is not in the document]
Comment (martastain) [reply api/system/info.py:164]: I've added a set serializer as we didn't support it before. it will make sense in various places.

**Review by BigRoy — APPROVED** [review]
Code changes look good to me. Didn't test.

File: api/system/info.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (BigRoy) [reply api/system/info.py:164]: Does that also 'deserialize' back to the `set`? I suppose it does due to how the models read them?

**Pull request merged**